src/api/ipfs.ts

```diff
@@ -0,0 +1,57 @@
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const ipfsClient = require("ipfs-http-client");
```

Comment:
`ipfs-http-client` should already have types, could you try with import?

Reply:
yeah i tried `import ipfsClient from 'ipfs-http-client'` and when i tried to `yarn run build`, it gave me all kinds of weird errors saying certain files were missing and all this (which didn't happen when using `require`, which was also the way that they said to use it in their readme, fwiw)

Reply:
here's an example of one of the errors i would see:
```
node_modules/ipfs-http-client/dist/src/index.d.ts:11:112 - error TS2307: Cannot find module 'ipfs-core/src/components/bitswap/stat' or its corresponding type declarations.

11         stat: (options?: (import("ipfs-core-types").AbortOptions & HttpOptions) | undefined) => Promise<import("ipfs-core/src/components/bitswap/stat").BitswapStats>;
                                                                                                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```